File: 2432-NumberOfZeroFilledSubarrays/2432-NumberOfZeroFilledSubarrays.js
// Last updated: 26/12/2025, 15:58:53
/**
 * @param {number[]} nums
 * @return {number}
 */
var zeroFilledSubarray = function(nums) {
    let current = 0;
    let result = 0;

    for(var i =0; i< nums.length; i++){
        if(nums[i]===0){
            ++current
            result +=  current 
        }else{
            current=0;
        }
    }

    return result;
};



//[0,0,0,2,0,0]
//have a result variable as 0
//multiplyerFactor = 1
//maxContinoutsZeros as another varaibale = 0
//start moving across
//if current number is zero then add maxContinoutsZeros++ * multiplyerFactor
//else set maxContinoutsZeros back to 0
//return result

File: 2432-NumberOfZeroFilledSubarrays/2432-NumberOfZeroFilledSubarrays.java
// Last updated: 23/04/2026, 17:59:00
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long count = 0, streak = 0;

        for(int n : nums){
            streak = (n ==0) ? (streak +1) : 0;
            count += streak;
        }

        return count;
    }

}